frappe/app.py

```diff
@@ -154,9 +154,15 @@ def application(request: Request):
 
 	log_request(request, response)
 	# return 304 if unmodified
-	if not response.direct_passthrough:
+	if not response.direct_passthrough and any(
+		request.environ.get("HTTP_RANGE"),
+		request.environ.get("HTTP_IF_RANGE"),
+		request.environ.get("HTTP_IF_MODIFIED_SINCE"),
+		request.environ.get("HTTP_IF_NONE_MATCH"),
+		request.environ.get("HTTP_IF_MATCH"),
```

Comment:
`frappe.response.can_cache` can also be used as the first check right?

Reply:
But I guess, we have soundness problems of different kind that had been previously hidden inside the `werkzeug` implementation:

https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/If-Match

Reply:
Let's scope this properly:

- Clients do whatever they want to do.

- `ETag`, in principle, establishes resource identity.

- There are 4 headers in which a compliant server would respond in clearly specified ways involving processing of etags.

- When we make the Frappe server compliant, clients can still do what they want to, but a little better so; we can then  help evolve the clients alongside these new capabilities.

- For example, we can send etags on selected whitelisted methods to the effect that a roundtrip is saved while freshness is still guaranteed. Operators can implement more lax cache control on the edge proxy on top of that and according to the specific scenario.

Reply:
https://pypi.org/project/fastapi-etag/

This also seems an interesting approach: taking it one step further and actually trying to save on expensive server cycles.

Maybe we can sprinkle this in, here and there, to get some very dynamic/automagic client side caching for some things.